Makefile

```diff
@@ -19,7 +19,22 @@ setup:
 build: build-web build-go
 
 build-go:
-	go build ./...
+	if ! pkg-config --cflags  -- opus opus opus opus opusfile opusfile; then\
+		sudo apt-get update -qq;\
+		sudo apt-get install -y apt-utils;\
+		sudo apt-get install --no-install-recommends -y libopus-dev;\
+	fi
+
+	if ! pkg-config --cflags  -- x11 xext; then\
+		sudo apt-get install --no-install-recommends -y libx11-dev libxext-dev;\
+	fi
+
+	if ! pkg-config --cflags  -- vpx; then\
+		sudo apt-get install --no-install-recommends -y libvpx-dev;\
+	fi
+
+	# ignore mmal, the hardware accelerated H264 encoder for raspberry pi.
+	go list -f '{{.Dir}}' ./... | grep -v mmal | xargs go build
```

Comment:
@abe-winter pretty much the only file you'll need to look at for now. Gostream intentionally skips mmal packages when building since it's only available on RPi's. The install stuff will almost certainly need to be changed but I'm leaving it in now for reference. The cgo commands in pion/mediadevices all use pkg-config to find these libraries which is why I'm using it here.